### ISSUE TITLE
No code sample 

### ISSUE THREAD

**Issue by HobDev** [crozone/LinuxSerialPort#7]
I am writing a Xamarin.Forms software targeting Xamarin.Mac to recognise "U.S Robotics Modem" attached to USB serial port. A window program is finding available ports by this code: string [] availableportnames=SerialPort.GetPortNames. But I am in total dark how to achieve this in LinuxSerialPort. Please add some sample code to help the users of this library.

**Comment by willhuo** [comment]
Pls add some sample code to help the user of thsi library

**Comment by crozone** [comment]
Hi, thanks for the issue. Sorry for the late response, I've just come back from vacation.

I'll get onto adding some code examples in the README, and some example code in the repository.

As to enumerating serial ports, I've created a new issue here: https://github.com/crozone/LinuxSerialPort/issues/8

TL;DR: `GetPortNames()` isn't implemented yet, but you can use the following code:

`IEnumerable<string> portPaths = Directory.GetFiles("/dev/", "tty*").Where(p => p.StartsWith("/dev/ttyS") || p.StartsWith("/dev/ttyUSB") || p.StartsWith("/dev/ttyACM" || p.StartsWith("/dev/ttyAMA");`

**EDIT**: I've just realized that you're on Macintosh. On OSX, the port names are again slightly different, and look like `/dev/tty.usb*`. I don't have a Mac available right now, but if you could run `ls /dev/tty*` and post the results, that will help me make the `GetPortNames()` method run on OSX as well.

**Comment by HobDev** [comment]
Hi, 
This is the result of running `ls /dev/tty*` on my Mac OS High Siera version 10.13.2
/dev/tty				/dev/ttysf
/dev/tty.Bluetooth-Incoming-Port	/dev/ttyt0
/dev/ttyp0				/dev/ttyt1
/dev/ttyp1				/dev/ttyt2
/dev/ttyp2				/dev/ttyt3
/dev/ttyp3				/dev/ttyt4
/dev/ttyp4				/dev/ttyt5
/dev/ttyp5				/dev/ttyt6
/dev/ttyp6				/dev/ttyt7
/dev/ttyp7				/dev/ttyt8
/dev/ttyp8				/dev/ttyt9
/dev/ttyp9				/dev/ttyta
/dev/ttypa				/dev/ttytb
/dev/ttypb				/dev/ttytc
/dev/ttypc				/dev/ttytd
/dev/ttypd				/dev/ttyte
/dev/ttype				/dev/ttytf
/dev/ttypf				/dev/ttyu0
/dev/ttyq0				/dev/ttyu1
/dev/ttyq1				/dev/ttyu2
/dev/ttyq2				/dev/ttyu3
/dev/ttyq3				/dev/ttyu4
/dev/ttyq4				/dev/ttyu5
/dev/ttyq5				/dev/ttyu6
/dev/ttyq6				/dev/ttyu7
/dev/ttyq7				/dev/ttyu8
/dev/ttyq8				/dev/ttyu9
/dev/ttyq9				/dev/ttyua
/dev/ttyqa				/dev/ttyub
/dev/ttyqb				/dev/ttyuc
/dev/ttyqc				/dev/ttyud
/dev/ttyqd				/dev/ttyue
/dev/ttyqe				/dev/ttyuf
/dev/ttyqf				/dev/ttyv0
/dev/ttyr0				/dev/ttyv1
/dev/ttyr1				/dev/ttyv2
/dev/ttyr2				/dev/ttyv3
/dev/ttyr3				/dev/ttyv4
/dev/ttyr4				/dev/ttyv5
/dev/ttyr5				/dev/ttyv6
/dev/ttyr6				/dev/ttyv7
/dev/ttyr7				/dev/ttyv8
/dev/ttyr8				/dev/ttyv9
/dev/ttyr9				/dev/ttyva
/dev/ttyra				/dev/ttyvb
/dev/ttyrb				/dev/ttyvc
/dev/ttyrc				/dev/ttyvd
/dev/ttyrd				/dev/ttyve
/dev/ttyre				/dev/ttyvf
/dev/ttyrf				/dev/ttyw0
/dev/ttys0				/dev/ttyw1
/dev/ttys000				/dev/ttyw2
/dev/ttys1				/dev/ttyw3
/dev/ttys2				/dev/ttyw4
/dev/ttys3				/dev/ttyw5
/dev/ttys4				/dev/ttyw6
/dev/ttys5				/dev/ttyw7
/dev/ttys6				/dev/ttyw8
/dev/ttys7				/dev/ttyw9
/dev/ttys8				/dev/ttywa
/dev/ttys9				/dev/ttywb
/dev/ttysa				/dev/ttywc
/dev/ttysb				/dev/ttywd
/dev/ttysc				/dev/ttywe
/dev/ttysd				/dev/ttywf
/dev/ttyse

**Comment by crozone** [comment]
@HobDev Thanks. Apparently on OSX there are also "call-up" devices that look like `/dev/cu.*`. Could you please run `ls /dev/cu.*` and post the results in the new issue? https://github.com/crozone/LinuxSerialPort/issues/8

Call-up devices are only slightly different to ttys, and it's possible that serial devices are one of the only things to create them. If this is true, `GetPortNames()` could be implemented by just returning `/dev/cu*`.

**Comment by crozone** [comment]
Fixed by https://github.com/crozone/LinuxSerialPort/commit/a8e0a7b4712392df432e42b82918368ac72e56fa
